includes/class-wc-gateway-ppec-checkout-handler.php

```diff
@@ -513,12 +513,14 @@ private function update_customer_addresses_from_paypal( $token ) {
 		$customer->set_billing_state( $billing_details['state'] );
 		$customer->set_billing_country( $billing_details['country'] );
 
-		$customer->set_shipping_address( $shipping_details['address_1'] );
-		$customer->set_shipping_address_2( $shipping_details['address_2'] );
-		$customer->set_shipping_city( $shipping_details['city'] );
-		$customer->set_shipping_postcode( $shipping_details['postcode'] );
-		$customer->set_shipping_state( $shipping_details['state'] );
-		$customer->set_shipping_country( $shipping_details['country'] );
+		if ( ! empty( $shipping_details ) ) {
+			$customer->set_shipping_address( $shipping_details['address_1'] );
```

Comment:
Are we certain all these keys will be present? Would this be a good place to use wp_parse_args to make sure each key has a default?